### PR TITLE
Junit testing

### PR DESCRIPTION
**Pull request by peterbae**
Moved JDBCEncryptionDecryption AE tests into Junit. Ran Appveyor and Travis tests against my local dev branch to ensure that the tests passed.

### CODE REVIEW

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/AESetup.java [hunk 1]
```diff
@@ -42,22 +43,29 @@
...
-    static String javaKeyStoreInputFile = "JavaKeyStore.txt";
-    static String keyStoreName = "MSSQL_JAVA_KEYSTORE";
-    static String jksName = "clientcert.jks";
+    static final String javaKeyStoreInputFile = "JavaKeyStore.txt";
+    static final String keyStoreName = "MSSQL_JAVA_KEYSTORE";
+    static final String jksName = "clientcert.jks";
+    static final String cmkName = "JDBC_CMK";
+    static final String cekName = "JDBC_CEK";
+    static final String secretstrJks = "password";
+	static final String numericTable = "JDBCEncrpytedNumericTable";
```
Comment (v-afrafi) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/AESetup.java]: can you run format on all of the file?

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/AESetup.java [hunk 2]
```diff
@@ -173,7 +425,18 @@ static void createCEK(SQLServerColumnEncryptionKeyStoreProvider storeProvider) t
...
         stmt.execute(cekSql);
     }
-
+    
+    /**
+     * Drop all tables that are in use by AE
+     * @throws SQLException
+     */
+	static void dropTables() throws SQLException {
+		stmt.executeUpdate("if object_id('" + numericTable + "','U') is not null" + " drop table " + numericTable);
```
Comment (v-afrafi) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/AESetup.java]: let's use Utils.dropTableIfExists

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/JDBCEncryptionDecryptionTest.java [hunk 3]
```diff
@@ -7,66 +7,505 @@
...
+	private LinkedList<byte[]> byteValuesNull = null;
+	
+	private LinkedList<Object> dateValues = null;
+    
     /**
-     * Test encryption and decryption of numeric values
-     * 
-     * @throws Exception
-     * @throws TestAbortedException
+     * Junit test case for char set string for string values
```
Comment (v-afrafi) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/JDBCEncryptionDecryptionTest.java]: same for this file. let's run format

File: src/main/java/com/microsoft/sqlserver/jdbc/Util.java [hunk 4]
```diff
@@ -970,7 +970,7 @@ static synchronized boolean checkIfNeedNewAccessToken(SQLServerConnection connec
 
     // if driver is for JDBC 42 and jvm version is 8 or higher, then always return as SQLServerPreparedStatement42,
     // otherwise return SQLServerPreparedStatement
-    static boolean use42Wrapper() {
+    public static boolean use42Wrapper() {
```
Comment (v-afrafi) [inline src/main/java/com/microsoft/sqlserver/jdbc/Util.java]: Could you check using System.getProperty("java.specification.version"); and skipping the test if java 7 would solve the issue?

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/AESetup.java (hunk 1 above)
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/AESetup.java]: done.

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/AESetup.java (hunk 2 above)
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/AESetup.java]: done.

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/JDBCEncryptionDecryptionTest.java (hunk 3 above)
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/JDBCEncryptionDecryptionTest.java]: done.

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/JDBCEncryptionDecryptionTest.java [hunk 5]
```diff
@@ -7,66 +7,505 @@
...
+     * @throws SQLException
+     */
+    @Test
+	public void testChar_SpecificSetter() throws SQLException {
```
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/JDBCEncryptionDecryptionTest.java]: done.

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/JDBCEncryptionDecryptionTest.java [hunk 6]
```diff
@@ -77,53 +516,1905 @@ static void dropAll() throws SQLServerException, SQLException {
...
+		// varbinary(8000)
+		for (int i = 13; i <= 15; i++) {
+			if (null == byteValues) {
+				pstmt.setObject(i, null, java.sql.Types.BINARY);
+			} else {
+				pstmt.setObject(i, byteValues.get(4));
+			}
+		}
+
+		pstmt.execute();
```
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/JDBCEncryptionDecryptionTest.java]: I created Util.close(ResultSet, Statement, Connection) that employs good practice for closing those 3 objects, and applied it to the test cases.

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/RandomData.java [hunk 7]
```diff
@@ -0,0 +1,655 @@
+package com.microsoft.sqlserver.jdbc.AlwaysEncrypted;
+import java.math.BigDecimal;
```
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/RandomData.java]: yep. randomData and Util were originally for AE only, but I think they can be used elsewhere as well - i moved both of them to util package.

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/RandomData.java [hunk 8]
```diff
@@ -0,0 +1,655 @@
...
+
+	private static String specicalCharSet = "ÀÂÃÄËßîðÐ";
+	private static String normalCharSet = "1234567890-=!@#$%^&*()_+qwertyuiop[]\\asdfghjkl;'zxcvbnm,./QWERTYUIOP{}|ASDFGHJKL:\"ZXCVBNM<>?";
+
+	private static String unicodeCharSet = "♠♣♥♦林花謝了春紅太匆匆無奈朝我附件为放假哇额外放我放问역사적으로본래한민족의영역은만주와연해주의일부를포함하였으나会和太空特工我來寒雨晚來風胭脂淚留人醉幾時重自是人生長恨水長東ྱོགས་སུ་འཁོར་བའི་ས་ཟླུུམ་ཞིག་ལ་ངོས་འཛིན་དགོས་ཏེ།ངག་ཕྱོαβγδεζηθικλμνξοπρστυφχψ太陽系の年齢もまた隕石の年代測定に依拠するので";
+
+	private static String numberCharSet = "1234567890";
+    private static String numberCharSet2 = "123456789"; 
+
+	//-------------------numeric types--------------------------------------------
```
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/RandomData.java]: done.

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/Util.java [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+		if (null == stmtColEncSetting)
+		{
+			return ((SQLServerConnection) connection).prepareCall(sql);
+		}
+		else
+		{
+			return ((SQLServerConnection) connection).prepareCall(sql, ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY, connection.getHoldability(), stmtColEncSetting);
+		}
+	}
+	/*
```
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/Util.java]: done.

File: src/test/java/com/microsoft/sqlserver/testframework/AbstractTest.java [hunk 10]
```diff
@@ -67,7 +67,8 @@ public static void setup() throws Exception {
         applicationKey = getConfiguredProperty("applicationKey");
         keyIDs = getConfiguredProperty("keyID", "").split(";");
 
-        connectionString = getConfiguredProperty("mssql_jdbc_test_connection_properties");
+        connectionString = getConfiguredProperty("mssql_jdbc_test_connection_properties")
```
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/testframework/AbstractTest.java]: done. I moved it to AESetup.java.

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/AESetup.java [hunk 11]
```diff
@@ -42,22 +43,29 @@
...
+    static final String jksName = "clientcert.jks";
+    static final String cmkName = "JDBC_CMK";
+    static final String cekName = "JDBC_CEK";
+    static final String secretstrJks = "password";
+	static final String numericTable = "JDBCEncrpytedNumericTable";
+	static final String charTable = "JDBCEncrpytedCharTable";
+	static final String binaryTable = "JDBCEncrpytedBinaryTable";
+	static final String dateTable = "JDBCEncrpytedDateTable";
+	static final String uid = "171fbe25-4331-4765-a838-b2e3eea3e7ea";
+	static final String uid2 = "171fbe25-4331-4765-a838-b2e3eea3e7eb";
```
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/AESetup.java]: Nope, I removed it.

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/AESetup.java [hunk 12]
```diff
@@ -70,23 +78,24 @@ static void setUpConnection() throws TestAbortedException, Exception {
 
         readFromFile(javaKeyStoreInputFile, "Alias name");
         con = (SQLServerConnection) DriverManager.getConnection(connectionString);
```
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/AESetup.java]: done.

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/AESetup.java [hunk 13]
```diff
@@ -70,23 +78,24 @@ static void setUpConnection() throws TestAbortedException, Exception {
...
-        stmt = con.createStatement();
+        stmt = (SQLServerStatement) con.createStatement();
         Utils.dropTableIfExists(numericTable, stmt);
```
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/AESetup.java]: done.

File: src/main/java/com/microsoft/sqlserver/jdbc/Util.java (hunk 4 above)
Comment (peterbae) [inline src/main/java/com/microsoft/sqlserver/jdbc/Util.java]: Checking for java 7 only doesn't solve the issue - the tests being done on github uses java 8, but does not support jdbc 4.2 - we need to check for both.

**Pull request merged**